chainer/training/extension.py

```diff
@@ -0,0 +1,109 @@
+PRIORITY_WRITER = 300
+PRIORITY_EDITOR = 200
+PRIORITY_READER = 100
+
+
+class Extension(object):
+
+    """Base class of trainer extensions.
+
+    Extension of :class:`Trainer` is a callable object that takes the trainer
+    object as the argument. It also provides some default configurations as its
+    attributes, e.g. the default trigger and the default priority. This class
+    provides a set of typical default values for these attributes.
+
+    There are two ways to define users' own extensions: inheriting this class,
+    or decorating closures by :func:`make_extension`. Decorator can slightly
+    reduce the overhead and is much easier to use, while this class provides
+    more flexibility (for example, it can have methods to configure the
+    behavior).
+
+    Attributes:
```

Comment:
I am not familiar with the convention of docstring but is it OK to add class variables to the Attributes entry?